pilot/pkg/xds/eds_test.go

```diff
@@ -430,6 +503,31 @@ func testEndpoints(expected string, cluster string, adsc *adsc.ADSC, t *testing.
 	t.Fatalf("Expecting %s got %v", expected, found)
 }
 
+// Verify server sends the tunneled endpoints.
+// nolint: unparam
+func testTunnelEndpoints(expectIP string, expectPort uint32, adsc *adsc.ADSC, t *testing.T) {
+	t.Helper()
+	cluster := "outbound|8080||eds.test.svc.cluster.local"
+	allClusters := adsc.GetEndpoints()
+	cla, f := allClusters[cluster]
+	if !f || len(cla.Endpoints) == 0 {
+		t.Fatalf("No lb endpoints for %v, %v", cluster, adsc.EndpointsJSON())
+	}
+	var found []string
+	for _, lbe := range cla.Endpoints {
```

Comment:
nit: this loop can be replaced by `xdstest.ExtractEndpoints`

Reply:
well not entirely, that gives a list of `[]string{ip:port}` though

Reply:
Endpoint, lbEndpoint, localityLbEndpoint, istioEndpoint are all legit `endpoint`